integrations/tensorflow/e2e/slim_vision_models/iree_slim_vision_test_suite.bzl

```diff
@@ -0,0 +1,155 @@
+        the backend to use as a source of truth for the expected output results.
+      failing_configurations:
+        an iterable of dictionaries with the keys `models` and `backends`. Each
+        key points to a string or iterable of strings specifying a set of models
+        and backends that are failing.
+      tags:
+        tags to apply to the test. Note that as in standard test suites, manual
+        is treated specially and will also apply to the test suite itself.
+      deps:
+        test dependencies.
+      size:
+        size of the tests. Default: "large".
+      python_version:
+        the python version to run the tests with. Uses python3 by default.
+      **kwargs:
+        any additional arguments that will be passed to the underlying tests and
+        test_suite.
+    """
+    failing_set = sets.make([])
+    if failing_configurations != None:
+        # Parse failing configurations.
+        for configuration in failing_configurations:
+            # Normalize configuration input.
+            # {backend: "iree_llvmjit"} -> {backend: ["iree_llvmjit"]}
+            for key, value in configuration.items():
+                if type(value) == type(""):
+                    configuration[key] = [value]
+
+            for model_tag in configuration["model_tags"]:
+                for backend in configuration["backends"]:
+                    sets.insert(failing_set, (model_tag, backend))
+
+    tests = []
+    for model_tag in model_tags:
+        for backend in backends:
+            # Check if this is a failing configuration.
+            failing = sets.contains(failing_set, (model_tag, backend))
+
+            # Append "_failing" to name if this is a failing configuration.
+            test_name = name if not failing else name + "_failing"
+            test_name = "{}_{}__{}__{}".format(
+                test_name,
+                model_tag,
+                reference_backend,
+                backend,
+            )
+            tests.append(test_name)
+
+            args = [
+                "--model_tag={}".format(model_tag),
+                "--reference_backend={}".format(reference_backend),
+                "--target_backends={}".format(backend),
+            ]
+
+            # TODO(GH-2175): Simplify this after backend names are
+            # standardized.
+            # "iree_<driver>" --> "<driver>"
+            driver = backend.replace("iree_", "")
+            if driver == "llvmjit":
+                driver = "llvm"
+            py_test_tags = ["driver={}".format(driver)]
+            if tags != None:  # `is` is not supported.
+                py_test_tags += tags
+
+            # Add additional tags if this is a failing configuration.
+            if failing:
+                py_test_tags += [
+                    "failing",  # Only used for test_suite filtering below.
+                    "manual",
+                    "nokokoro",
+                    "notap",
+                ]
```

Comment:
This is going to end up with potentially duplicated tags if these tags are already present. Not sure if that creates any problems, but maybe we should check?

Reply:
Good point. This would be a problem with `iree_vision_test_suite` too, so I'm guessing that it doesn't cause any problems. I can't find anything mentioning it internally or externally, but I'm not the best sleuth.

Reply:
Let's leave it consistent for now and we can investigate more if we care.

Reply:
SGTM.